app/database_etl/location_utils/location_util_functions.py

```diff
@@ -124,29 +124,29 @@ def row_in_feature_layer(row: pd.Series, feature_layer: FeatureLayer) -> bool:
 # Get pin latitude and longitude for a record
 # Param geo_dfs = dictionary of city, state, and country DFs
 def get_record_coordinates(record: pd.Series, geo_dfs: dict) -> Tuple:
-    pin_regions = [record['country']]
-    pin_region_type = 'country' if record['country'] is not None else ''
+    pin_lat = None
+    pin_lng = None
 
-    for region_type in ['state', 'city']:
+    for region_type in ['country', 'state', 'city']:
```

Comment:
I'm a bit confused by the design pattern here. 

What I might have expected this to look like is something like the following. In essence, this code looks from small to large to see whether it can find a latlong for the most granular geography type, and if not, steps back one broader.

```
for region_type in ['city', 'state', 'country']: # from small to large
  if not record[region_type]:
    continue
  lat, lng = get_latlong(set(record[region_type], record['country']))
  # using the two in combination to ensure that you don't get e.g., Calgary in Scotland instead of Calgary in Canada
  # with some sort of deduplication if region_type is country
  if lat and lng: 
    break
```

Reply:
I kind of also agree with this -- I'm confused why we are averaging the pin lat lngs from the city, state and country? Wouldn't we want the most accurate lat long value if it's avail? i.e. if city is available, take city, otherwise take state, otherwise take country? maybe I don't understand the OG logic too well myself either

Reply:
Instead of going from granular to least granular and backtracking if necessary, this logic goes from least granular to most granular and breaks when we can no longer go more granular. That way no backtracking is necessary (which I would find confusing)

Reply:
We are averaging in the case where we have multiple cities or multiple states. We made a decision that if say we had a study conducted in multiple cities, the pin would show up in the center of those cities.

Reply:
@rahularoradfs using a country and city to narrow down on a location isn't enough. A country can have multiple of the same cities if they're they're different states. The problem is right now our airtable schema is not robust enough to be able to infer these relationships (we simply store lists of cities and states for each record).

Why don't we consider improving this in our airtable refactor?